api/routers/shortener.py

```diff
@@ -21,15 +27,31 @@ def create_shortened_url(
 	db_session: Session = Depends(get_db_session),
 	original_url: str = Form(...)):
 	try:
-		short_url = return_short_url(original_url, db_session)
-		data = {
-			"short_url": short_url,
-			"url": "http://localhost:8000/" + short_url,
-			"button": "Shorten now"
-		}
+		# Check to see if the url confronts to a valid format. If not then display error.
+		if (not is_valid_url(original_url)):
+			data = {
+				"error": "Unable to shorten that link. It is not a valid url.",
+				"url": original_url
+			}
+		# Else if the domain is not allowed, display error and link to GC Forms page
+		elif (not is_domain_allowed(original_url, db_session)):
+			forms_url = os.getenv("FORMS_URL")
+			data = {
+				"error": "Your url is not a registered Government of Canada domain in our system.",
+				"form_url": forms_url,
+				"url": original_url
+			}
+		# Else, we are all good to shorten!
+		else:
+			short_url = return_short_url(original_url, db_session)
+			SHORTENER_DOMAIN = os.getenv("SHORTENER_DOMAIN") or None
+			data = {
+				"short_url": short_url,
+				"url": SHORTENER_DOMAIN + short_url,
+			}
 	except Exception as err:
 		data = {
-			"error" : "Error in processing shortened url"
+			"error": f"Error in processing shortened url {err}"
```

Comment:
Whilst I think printing errors is useful in dev; could printing the exception accidentally leak information when run in a production environment?

Reply:
It could. I will add an issue ticket in github to make sure we "secure" our api including removing such messages. I think for now we could leave it, but think about adding an env variable that would have production/dev environment.

Reply:
Issue is submitted [here](https://github.com/cds-snc/url-shortener/issues/14).